pkg/metrics/statsd/statsd_test.go

```diff
@@ -83,3 +86,161 @@ func TestEmitFloat(t *testing.T) {
+		c2.SetReadDeadline(time.Now().Add(5 * time.Second))
+		m := ""
+		//fmt.Println("Read")
+		n, _, err := c2.ReadFrom(buf)
+		if err != nil {
+			//fmt.Println(err.Error())
+			//m = err.Error()
+			m = err.Error()
+		} else {
+			m = string(buf[:n])
+		}
+
+		c <- m
+	}()
+
+	go s.run()
+	s.EmitGauge("tests.test_key", 42, map[string]string{"key": "value"})
+	m := <-c
+	if m != `{"Metric":"tests.test_key","Account":"*","Namespace":"*","Dims":{"hostname":"test-host","key":"value","location":"eastus"},"TS":"0001-01-01T00:00:00.000"}:42|g`+"\n" {
+		t.Error(errors.New(m))
+	}
+
+}
+
+func TestGetConnectionDetails(t *testing.T) {
+
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	for _, tt := range []struct {
+		name             string
+		set              bool
+		isLocalDev       bool
+		socketstring     string
+		protocol         string
+		connectionstring string
+		wantError        bool
+	}{
+		{
+			name:         "Random string",
+			set:          true,
+			socketstring: "randomstring234234809$#54ew5",
+			wantError:    true,
+		},
+		{
+			name:             "Old Bevaviour / production mode",
+			set:              false,
+			protocol:         "unix",
+			connectionstring: "/var/etw/mdm_statsd.socket",
+			wantError:        false,
+		},
+		{
+			name:             "Old Bevaviour / localdev mode",
+			set:              false,
+			isLocalDev:       true,
+			protocol:         "unix",
+			connectionstring: "mdm_statsd.socket",
+			wantError:        false,
+		},
+		{
+			name:         "Empty env variable",
+			set:          true,
+			socketstring: "",
+			wantError:    true,
+		},
+		{
+			name:             "Valid UDP env variable",
+			set:              true,
+			socketstring:     "udp:127.0.0.1:9000",
+			protocol:         "udp",
+			connectionstring: "127.0.0.1:9000",
+			wantError:        false,
+		},
+		{
+			name:             "Valid Unix domain socket env variable",
+			set:              true,
+			socketstring:     "unix:test.socket",
+			protocol:         "unix",
+			connectionstring: "test.socket",
+			wantError:        false,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			fmt.Println(tt.name)
+			env := mock_env.NewMockInterface(controller)
+
+			if !tt.set {
+				if tt.isLocalDev {
+					env.EXPECT().IsLocalDevelopmentMode().Return(true)
+				} else {
+					env.EXPECT().IsLocalDevelopmentMode().Return(false)
+				}
+			}
+
+			s := &statsd{
+				env: env,
+			}
+
+			os.Unsetenv(STATSD_SOCKET_ENV)
+			if tt.set {
+				os.Setenv(STATSD_SOCKET_ENV, tt.socketstring)
+			}
+
+			p, c, err := s.getConnectionDetails()
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+
+			if tt.wantError {
+				if err == nil {
+					t.Fail()
+				}
+			} else {
+				if err != nil && tt.wantError == false {
+					fmt.Println(err.Error())
+					t.Fail()
+				} else if p != tt.protocol {
+					t.Fail()
+				} else if c != tt.connectionstring {
+					t.Fail()
+				}
```

Comment:
why not use `p != tt.protocol || c != tt.connectionstring` here , since there is no difference in handling ?

Reply:
My first version of the code actually combined all three cases in one if statement, before I changed it to this, which I feel is more legible and easier to troubleshoot/debug/set breakpoint. What u think ?

Reply:
ok , that's fair.

Reply:
I think we can drop the `else`s 
```suggestion
				if err != nil && tt.wantError == false {
					fmt.Println(err.Error())
					t.Fail()
				} 
				if p != tt.protocol {
					t.Fail()
				} 
				if c != tt.connectionstring {
					t.Fail()
				}
```

Reply:
Dropped